### PR TITLE
Fix setup of pod routes

### PR DESCRIPTION
**Pull request by AlainRoy**
There was a race condition in setting up the pod routes that could
result in them not working.

When we set up the routes, we wait for the Linux bridge (cbr0) to have
an IP address. We then find the subnet it's on (e.g. 10.244.0.0/24)
and set up routes based on that subnet.

However, when the bridge comes up, it comes up in stages, and we'll
see two IP addresses. The first IP address is in the 172.x.x.x
network, and the second one will be the real network. The code would
sometimes use the 172 network and sometimes the real network.

This update waits for the real address, and now pod networking works:
all nodes can access all pod and cluster IP addresses.

### CODE REVIEW

File: cluster/photon-controller/util.sh [hunk 1]
```diff
@@ -560,11 +560,20 @@ function wait-minion-apis {
...
+        # because it's both simpler and more clear what's happening.
         try-until-success-ssh ${KUBE_NODE_IP_ADDRESSES[$node]} \
-            "Waiting for cbr0 bridge on ${NODE_NAMES[$node]}"  \
-            'sudo ifconfig cbr0 | grep -oP "inet addr:\K\S+"'
+            "Waiting for cbr0 bridge on ${NODE_NAMES[$node]} to have an address"  \
+            'sudo ifconfig cbr0  | grep -oP "inet addr:\K\S+"'
+
+        try-until-success-ssh ${KUBE_NODE_IP_ADDRESSES[$node]} \
+            "Waiting for cbr0 bridge on ${NODE_NAMES[$node]} to have correct address"  \
+            'sudo ifconfig cbr0  | grep -oP "inet addr:\K\S+" | grep -v  "^172."'
```
Comment (faijaz) [inline cluster/photon-controller/util.sh]: Isn't it very hacky way to fix it? Is there any other way we can figure out the real network?

Comment (AlainRoy) [inline cluster/photon-controller/util.sh]: It's a little bit hacky, but it works in almost all cases. I don't know what a perfect fix is. I'm thinking about it and will do something better in the future. 

The reason this almost always works is two-fold:
* The cbr0 bridge always comes up on a 172.x.x.x address, at least with how we're bringing it up. It's in the RFC 1918 private space for IP addresses.
* The normal configuration for Kubernetes is to not use the 172.x.x.x addresses for the final address.

So I agree with you, but this is a reasonable approach for now.

**Comment by faijaz** [comment]
Good to check-in as temporary fix for now. But I believe we should look at some permanent fix in future.

File: cluster/photon-controller/util.sh (hunk 1 above)
Comment (imkin) [inline cluster/photon-controller/util.sh]: We could merge these two blocks together?

**Pull request merged**